test/unit/components/common/cgTable/cgTable_spec.js

```diff
@@ -0,0 +1,139 @@
+          isSelected: false
+        },
+        {
+          samples: 390,
+          negatives: 390,
+          positives: 0,
+          acronym: "BLCA",
+          name: "bladder urothelial carcinoma",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 976,
+          negatives: 976,
+          positives: 0,
+          acronym: "BRCA",
+          name: "breast invasive carcinoma",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 193,
+          negatives: 193,
+          positives: 0,
+          acronym: "CESC",
+          name: "cervical & endocervical cancer",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 36,
+          negatives: 36,
+          positives: 0,
+          acronym: "CHOL",
+          name: "cholangiocarcinoma",
+          isLoading: false,
+          isSelected: true
+        }
+      ];
+
+      parentScope.columns = [
+        {
+          name: "Symbol",
+          id: "symbol",
+          isSortable: true
+        },
+        {
+          name: "Name",
+          id: "name",
+          isSortable: true
+        },
+        {
+          name: "Entrez ID",
+          id: "entrezgene",
+          isSortable: true
+        },
+        {
+          name: "Score",
+          id: "score",
+          isSortable: true
+        }
+      ];
+
+      element = angular.element(`
+          <cg-table
+            data="data"
+            columns="columns"
+          ></cg-table>        
+        `);
+
+      $compile(element)(parentScope);
+
+      controller = element.controller("cgTable");
+
+      parentScope.$digest();
+    })
+  );
+
+  it("displays the correct amount of columns", () => {
+    let columnCells = angular.element(element[0].querySelectorAll("th"));
+    expect(columnCells.length).toEqual(parentScope.columns.length + 1);
+  });
+
+  it("displays the correct amount of rows", () => {
+    let tableRows = angular.element(element[0].querySelectorAll("tbody tr"));
+    expect(tableRows.length).toEqual(parentScope.data.length);
+  });
+
+  it("detects if all results are selected", () => {
+    expect(controller.areAllResultsSelected()).toBe(false);
+  });
+
+  it("selects all results", () => {
+    controller.selectAllResults(true);
+    const areResultsAllSelected = parentScope.data.every(
+      result => result.isSelected === true
+    );
+    expect(areResultsAllSelected).toBe(true);
+  });
+
+  it("unselects all results", () => {
+    controller.selectAllResults(false);
+    const areResultsAllSelected = parentScope.data.every(
+      result => result.isSelected === true
+    );
+    expect(areResultsAllSelected).toBe(false);
+  });
+
+  it("sets sortType and sortReverse correctly when sorting", () => {
+    controller.sortColumn("Name");
+    expect(controller.sortType).toEqual("Name");
+    expect(controller.sortReverse).toBe(true);
+  });
```

Comment:
Hmm. Is it really desired behavior that the sort order reverses *every time* the sort column is changed? I could see this happening if you click *again* on the currently-sorted column (and that is pretty standard), but that is not the behavior coming from this test nor from the implementation in `cgTable.component.js`, line 46.

If flipping the sort order each time is really what should happen, then this test (and the component) works well. If not, this is a good opportunity to fix a sorting bug.